Celeste.Mod.mm/Patches/Monocle/Tracker.cs

```diff
@@ -34,16 +34,40 @@ public class TrackedAsAttribute : Attribute {
     }
 
     class patch_Tracker : Tracker {
-        [MonoModIgnore]
-        public static extern List<Type> GetSubclasses(Type type);
+        // A temporary cache to store the results of FakeAssembly.GetFakeEntryAssemblies
+        // Set to null outside of Initialize.
+        private static Type[] _temporaryAllTypes;
+
+        private static Type[] GetAllTypesUncached() => FakeAssembly.GetFakeEntryAssembly().GetTypesSafe();
+        
+        [MonoModReplace]
+        private static List<Type> GetSubclasses(Type type) {
+            bool shouldNullOutCache = _temporaryAllTypes is null;
+            _temporaryAllTypes ??= GetAllTypesUncached();
+            
+            List<Type> subclasses = new();
+            foreach (Type otherType in _temporaryAllTypes)
+            {
+                if (type != otherType && type.IsAssignableFrom(otherType))
+                    subclasses.Add(otherType);
+            }
```

Comment:
Could this be done with LINQ instead?  I'm not sure what the difference in allocations or stack size would be.

```cs
List<Type> subclasses = _temporaryAllTypes
    .Where(otherType => type != otherType && type.IsAssignableFrom(otherType)
    .ToList();
```

Reply:
Linq is always going to be slower for this, unfortunately